src/main/java/org/apache/maven/shared/utils/cli/Commandline.java

```diff
@@ -198,14 +200,13 @@ public void addArguments( String... line )
      */
     public void addEnvironment( String name, String value )
     {
-        //envVars.add( name + "=" + value );
         envVars.put( name, value );
     }
 
     /**
      * Add system environment variables.
      */
-    public void addSystemEnvironment()
+    private void addSystemEnvironment()
```

Comment:
This looks like an API breaking change

Reply:
Yes I know, this method is confusing because it is used internally in `getEnvironmentVariables`.
So when we switch to `4.0.0` I think that can break api in this point.
Now will be setShellEnvironmentInherited for achieve this feature.